### DISCUSSION TITLE
Disable the message: "The package *** provides CMake targets:"

### DISCUSSION THREAD

**Discussion by Zingam** [microsoft/vcpkg#27153] [Ideas]
I'd like to disable the messages manually: Disable the message: "The package *** provides CMake targets:"
Even better it would be if a target is used that this message is not displayed during subsequent "configurations".

This is particularly annoying when using vcpkg integration with VSCode's extensions.

```json
{
  "cmake.configureSettings": {
    "CMAKE_TOOLCHAIN_FILE": "[vcpkg root]/scripts/buildsystems/vcpkg.cmake"
  }
}
```

**Comment by autoantwort** [comment]
I think you can use `--no-print-usage`

**Comment by Zingam** [comment]
I failed to explain that I am talking about toolchain integration with CMake.

**Comment by autoantwort** [comment]
Then use `set(VCPKG_INSTALL_OPTIONS --no-print-usage)` before the `project` call (or pass `-DVCPKG_INSTALL_OPTIONS=--no-print-usage` when calling cmake)

**Reply by Zingam** [reply]
I'm using CMake presets + VSCode/cpptools. I tried your both suggestions without success.

```json
  "configurePresets": [
    {
      "name": "default",
      "hidden": true,
      "displayName": "Default Config",
      "description": "Default build using Ninja generator",
      "generator": "Ninja",
      "binaryDir": "${sourceDir}/../__build/${presetName}",
      "cacheVariables": {
        "VCPKG_INSTALL_OPTIONS": "--no-print-usage"
      },
      "environment": {},
      "toolchainFile": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
    },
```

This is the generated command with the second option, which looks like what you suggested:

> [proc] Executing command: /snap/cmake/current/bin/cmake -DVCPKG_INSTALL_OPTIONS=--no-print-usage -DCMAKE_TOOLCHAIN_FILE=/home/user/SDKs/vcpkg/scripts/buildsystems/vcpkg.cmake -S/home/user/Projects/FE-CPP/FE -B/home/hristo/Projects/FE-CPP/__build/default.ninja-multi -G "Ninja Multi-Config"

Output:

> [proc] Executing command: /snap/cmake/current/bin/cmake -DVCPKG_INSTALL_OPTIONS=--no-print-usage -DCMAKE_TOOLCHAIN_FILE=/home/user/SDKs/vcpkg/scripts/buildsystems/vcpkg.cmake -S/home/user/Projects/FE-CPP/FirstEngine -B/home/user/Projects/FE-CPP/__build/default.ninja-multi -G "Ninja Multi-Config"
[cmake] -- Running vcpkg install
[cmake] Detecting compiler hash for triplet x64-linux...
[cmake] All requested packages are currently installed.
[cmake] Restored 0 package(s) from /home/hristo/.cache/vcpkg/archives in 2.74 us. Use --debug to see more details.
[cmake] 
[cmake] Total elapsed time: 2.461 s
[cmake] cxxopts provides CMake targets:
[cmake] 
[cmake]     # this is heuristically generated, and may not be correct
[cmake]     find_package(cxxopts CONFIG REQUIRED)
[cmake]     target_link_libraries(main PRIVATE cxxopts::cxxopts)
[cmake] 
[cmake] The package fmt provides CMake targets:
[cmake] 
[cmake]     find_package(fmt CONFIG REQUIRED)
[cmake]     target_link_libraries(main PRIVATE fmt::fmt)
[cmake] 
[cmake]     # Or use the header-only version
[cmake]     find_package(fmt CONFIG REQUIRED)
[cmake]     target_link_libraries(main PRIVATE fmt::fmt-header-only)
[cmake] 
[cmake] tomlplusplus provides CMake targets:
[cmake] 
[cmake]     # this is heuristically generated, and may not be correct
[cmake]     find_package(tomlplusplus CONFIG REQUIRED)
[cmake]     target_link_libraries(main PRIVATE tomlplusplus::tomlplusplus)
[cmake] 
[cmake] -- Running vcpkg install - done
[cmake] -- The C compiler identification is GNU 12.1.0
[cmake] -- The CXX compiler identification is GNU 12.1.0
[cmake] -- Detecting C compiler ABI info
[cmake] -- Detecting C compiler ABI info - done
[cmake] -- Check for working C compiler: /usr/bin/cc - skipped
[cmake] -- Detecting C compile features
[cmake] -- Detecting C compile features - done
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Check for working CXX compiler: /usr/bin/c++ - skipped
[cmake] -- Detecting CXX compile features
[cmake] -- Detecting CXX compile features - done
[cmake] -- Found Vulkan: /home/user/SDKs/Vulkan/vulkansdk-linux-x86_64-1.3.211.0/1.3.211.0/x86_64/lib/libvulkan.so (found version "1.3.211") found components: glslc glslangValidator 
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: /home/user/Projects/FE-CPP/__build/default.ninja-multi

**Reply by autoantwort** [reply]
You have to wait for https://github.com/microsoft/vcpkg/pull/27209
